manager/src/main/java/io/gingersnapproject/json/JsonHelper.java

```diff
@@ -0,0 +1,38 @@
+package io.gingersnapproject.json;
+
+import java.util.List;
+
+import org.infinispan.commons.dataconversion.internal.Json;
+
+import io.gingersnapproject.search.QueryResult;
+import io.smallrye.mutiny.Uni;
+
+public final class JsonHelper {
+
+   private JsonHelper() {
+   }
+
+   public static Uni<String> json(Uni<QueryResult> queryResultUni) {
+      return queryResultUni.onItem().transform(queryResult -> {
+         // Json.make() seems to be not working:
+         // > doing it manually
+         Json result = Json.object();
+
+         result.set("hitCount", queryResult.hitCount());
+         result.set("hitCountExact", queryResult.hitCountExact());
+         result.set("hits", JsonHelper.make(queryResult.hits()));
+         result.set("hitsExacts", queryResult.hitsExacts());
+
+         return result.toString();
+      });
+   }
+
+   public static Json make(List<String> hits) {
+      Json array = Json.array();
+      for (String item : hits) {
+         // TODO this is very inefficient
```

Comment:
I don't think there is any other way to do this if you want a nested Json object. It needs to be parsed. Or are you thinking of creating a larger String array Json representation and calling to parse once?

Reply:
Maybe we could avoid parsing the JSON hits, building the entire QueryResult JSON manually from a StringBuilder. But in any case let's keep it as it is for now.